state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -758,3 +758,11 @@ impl<T: Config> NetworkEnvironmentProvider for Pallet<T> {
 		Self::network_environment()
 	}
 }
+
+pub struct NoUsedNonce<T: Config>(PhantomData<T>);
+
+impl<T: Config> ExecutionCondition for NoUsedNonce<T> {
+	fn is_satisfied() -> bool {
+		SolanaAvailableNonceAccounts::<T>::get().len() == 10
```

Comment:
I thought it was 7 😅 

Could we not use `SolanaUnavailableNonceAccounts::<T>::iter().next().is_none()`?

Reply:
It was originally 7 indeed. However, I did some optimizations to be able to bump it up to ten, as that is the bottleneck.

Reply:
Sorry, should been less indirect: we can't assume that it will always be 10. We've already change it once, we might change it again. So the implementation should not assume the number.